archaius2-core/src/main/java/com/netflix/archaius/config/polling/FixedPollingStrategy.java

```diff
@@ -44,17 +45,23 @@ public FixedPollingStrategy(long interval, TimeUnit units, boolean syncInit) {
         this.interval = interval;
         this.units    = units;
         this.syncInit = syncInit;
+
     }
     
     @Override
     public Future<?> execute(final Runnable callback) {
-        while (syncInit) {
+        int numRetries = MAX_NUM_RETRIES;
+        while (true) {
```

Comment:
This look looks a bit awkward in how the numRetries update and condition check are disconnected.  This might read better as
```java
while (syncInit) {
   try { 
      callback.run();
      break;
   }
   catch (Exception e) {
      if (--numRetries == 0) {
          return Futures.immediateFailure(e); 
      }
      try {
          units.sleep(interval);
      } catch (InterruptedException e1) {
          Thread.currentThread().interrupt();
          return Futures.immediateFailure(e);
      }
   }
}
```